docs/tech/compatibility-management.md

```diff
@@ -0,0 +1,77 @@
+
+The Api classes in BDK will be generated from these specifications and then be used in BDK services and 
+BDK authenticator (a special BDK service).
+
+## BDK compatibility based on the version of Symphony-api-spec
+Because the BDK used the Symphony-api-spec for generating the Api classes, then the compatibility between 
+BDK and SBE/Agent will depend on the version of Symphony-api-spec that the BDK is using and the compatibility between 
+the specs and the SBE/Agent.
+
+From the official version 2.0.0 of the BDK, we start using the version branch of the Symphony-api-spec 
+(20.9 for the BDK 2.0.0), it means that the version of the BDK will compatible with the SBE product version 20.9.0.
+
+## API change management
+
+Each BDK version will be tied to one specific product version corresponding to one version of api specifications, 
+and the version of supported SBE and Agent can be found in [Agent compatibilities](https://developers.symphony.com/restapi/docs/agent-compatibilities).
+
+| Version         | Supported by BDK |
+|-----------------|------------------|
+| 20.9 and before | 2.0.0            |
+
+Rationale of our change management policy is to be backward compatible on existing features
+(classes, methods or [configuration](../configuration.md) fields) whenever possible.
+
+## Deprecations and breaking changes
+
+For keeping the backward compatibility of the BDK, the BDK api classes will be generated from the xx-api-public-deprecated.yaml 
+file instead of the normal one. By doing this way, the generated APIs which are deprecated will not be completely removed.
+They are only annotated with `@Deprecated`. 
+
+Then, the BDK services which are using these deprecated APIs will be annotated with `@Deprecated` as well.
+Then the bot developers will be warned that these services are now deprecated and will be removed in the future without having any breaking changes.
+
+In case of a non-backward compatible change:
+* we will mark the configuration field,
+class or method as deprecated in a given `x.y.z` version.
+* Version `x.(y + 1).0` will start logging a `WARN` message if the class, method or configuration field is called.
+  Since deprecation messages are logged using a dedicated `com.symphony.bdk.deprecation` logger, logs can be disabled
+  by setting its log level to `ERROR`. See [log4j.properties](../../symphony-bdk-examples/bdk-core-examples/src/main/resources/log4j.properties)
+  or [logback.xml](../../symphony-bdk-examples/bdk-core-examples/src/main/resources/logback.xml) on how to configure this.
+* Deprecated items will be then removed in `x.(y + 2).0` version *only* if a replacement has been
+documented and is available.
+
+Deprecations and subsequent class, method or field removal will all be documented in
+[release notes](https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/releases).
+
+For all the new features, deprecations and breaking changes, they should be mentioned in this feature support matrix
```

Comment:
so in the release notes or in this table?
let's keep the release notes for now it will be simpler to maintain